crates/holochain_cascade/src/authority/test.rs

```diff
@@ -295,3 +294,108 @@ async fn get_agent_activity() {
 
     pretty_assertions::assert_eq!(result, expected);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn get_agent_activity_respects_withhold_publish() {
+    holochain_trace::test_run();
+    let db = test_dht_db();
+
+    let td = ActivityTestData::valid_chain_scenario(false);
+
+    for hash_op in td.agent_activity_ops.iter().cloned() {
+        fill_db(&db.to_db(), hash_op).await;
+    }
+    // Mark the most recent valid op as withheld
+    let last_op_hash = td.agent_activity_ops.last().unwrap().hash.clone();
+    db.write_async(move |txn| set_withhold_publish(txn, &last_op_hash))
+        .await
+        .unwrap();
+
+    let options = actor::GetActivityOptions {
+        include_valid_activity: true,
+        include_rejected_activity: false,
+        include_full_records: false,
+        ..Default::default()
+    };
+
+    let result = handle_get_agent_activity(
+        db.to_db().into(),
+        td.agent.clone(),
+        QueryFilter::new(),
+        (&options).into(),
+    )
+    .await
+    .unwrap();
+
+    let hashes_without_head = match &td.valid_hashes {
+        ChainItems::Hashes(hashes) => hashes.iter().take(hashes.len() - 1).cloned().collect(),
+        _ => unreachable!(),
+    };
+    let chain_head = match &td.valid_hashes {
+        ChainItems::Hashes(hashes) => hashes.iter().nth_back(1).cloned().unwrap(),
```

Comment:
?
```suggestion
        ChainItems::Hashes(hashes) => hashes.iter().first().cloned().unwrap(),
```

Reply:
In a list of 50 items, with 0 based indexing - `first_back(1)` is index 48, where `first` is index 0

I know it looks weird, but clippy shouted at me when I tried to do this a way that looked natural to me :D

Reply:
It's hard to understand why you'd want the second-to-last item and not the last item? Maybe the variable should be called something other than `chain_head`?